server/bin/pbench-tarball-stats.py

```diff
@@ -0,0 +1,196 @@
+                    else:
+                        tb_dt = None
+                    yield TarBallInfo(
+                        ctrl=c_entry.name, tb=entry.name, dt=tb_dt, stat=entry.stat()
+                    )
+
+
+def report_by_bucket(buckets):
+    for bucket_name, bucket_list in sorted(buckets.items()):
+        count = len(bucket_list)
+        ec2_cnt = 0
+        for item in bucket_list:
+            if item[0].startswith("EC2::"):
+                ec2_cnt += 1
+        pct_ec2 = (ec2_cnt / count) * 100.0
+        print(f"{bucket_name}: {count:7n} (%EC2 {pct_ec2:0.02f})")
+
+
+def main(options):
+    locale.setlocale(locale.LC_ALL, "en_US.UTF-8")
+    if not options.cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable",
+            file=sys.stderr,
+        )
+        return 1
+
+    try:
+        config = pbench.PbenchConfig(options.cfg_name)
+    except pbench.BadConfig as e:
+        print(f"{_NAME_}: {e}", file=sys.stderr)
+        return 1
+
+    archive = config.ARCHIVE
+    archive_p = os.path.realpath(archive)
+
+    if not archive_p:
+        print(
+            f"The configured ARCHIVE directory, {archive}, does not exist",
+            file=sys.stderr,
+        )
+        return 1
+
+    if not os.path.isdir(archive_p):
+        print(
+            f"The configured ARCHIVE directory, {archive}, is not a valid directory",
+            file=sys.stderr,
+        )
+        return 1
+
+    invalid_cnt = 0
+    invalid_size = 0
+    invalid = collections.defaultdict(list)
+
+    good_cnt = 0
+    good_size = 0
+    good = collections.defaultdict(list)
+
+    ec2_cnt = 0
+    rh_cnt = 0
+
+    by_year = collections.defaultdict(list)
+    by_month = collections.defaultdict(list)
+
+    start = pbench._time()
+
+    gen = gen_tar_balls(archive_p)
+
+    for tb_rec in gen:
+        if tb_rec.dt is None:
+            invalid[tb_rec.ctrl].append(tb_rec)
+            invalid_cnt += 1
+            invalid_size += tb_rec.stat.st_size
+        else:
+            good[tb_rec.ctrl].append(tb_rec)
+            good_cnt += 1
+            good_size += tb_rec.stat.st_size
+            if tb_rec.ctrl.startswith("EC2::"):
+                ec2_cnt += 1
+            else:
+                rh_cnt += 1
+            year = tb_rec.dt.year
+            by_year[year].append(tb_rec)
+            month = tb_rec.dt.month
+            ym = f"{year:4d}-{month:02d}"
+            by_month[ym].append(tb_rec)
+
+    time_sec = pbench._time() - start
+
+    print(f"Took {time_sec:0.2f} seconds")
+
+    print(f"{len(invalid.keys()):18n} controllers with bad tar balls")
+    print(f"{invalid_cnt:18n} bad tar balls, total count")
+    print(f"{invalid_size:18n} bad tar balls, total size")
+
+    print(
+        f"{len(good.keys()):18n} controllers with good tar balls (EC2 {ec2_cnt:n}, RH {rh_cnt:n})"
+    )
+    print(f"{good_cnt:18n} good tar balls, total count")
```

Comment:
Perhaps the EC2/RH breakdown belongs on line 181 rather than line 179.
(See comment on `server/bin/gold/test-5.2.txt` above.)